src/qt/addresstablemodel.cpp

```diff
@@ -52,16 +53,17 @@ struct AddressTableEntryLessThan
 };
 
 /* Determine address type from address purpose */
-static AddressTableEntry::Type translateTransactionType(const QString &strPurpose, bool isMine)
+static AddressTableEntry::Type translateTransactionType(wallet::AddressPurpose purpose, bool isMine)
 {
     AddressTableEntry::Type addressType = AddressTableEntry::Hidden;
     // "refund" addresses aren't shown, and change addresses aren't returned by getAddresses at all.
-    if (strPurpose == "send")
+    if (purpose == wallet::AddressPurpose::SEND) {
         addressType = AddressTableEntry::Sending;
-    else if (strPurpose == "receive")
+    } else if (purpose == wallet::AddressPurpose::RECEIVE) {
         addressType = AddressTableEntry::Receiving;
-    else if (strPurpose == "unknown" || strPurpose == "") // if purpose not set, guess
+    } else {
         addressType = (isMine ? AddressTableEntry::Receiving : AddressTableEntry::Sending);
+    }
```

Comment:
IIRC, don't we get all sorts of compiler goodness if we use `switch` with an enum, like compiler warnings for unhandled cases? Would be nice, in the event we ever add a new case to the enum (although seems unlikely for this specific use-case).

I always prefer using `switch` with enums, but feel free to ignore if you don't feel like retouching/don't share the same opinion.

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wswitch-enum , super nice and still warns for unhandled cases even when you have a default.

Reply:
Done as suggested